app/models/generates_notifications.rb

```diff
@@ -0,0 +1,13 @@
+module GeneratesNotifications
+    extend self
+
+    def generate_notifications_for(users, title:)
+        users.each do |user|
+          notification = Notification.new
+          notification.title = title
+          notification.source = self
```

Comment:
Is this the only reason why you have this a as a module? Why not pass it as a parameter and put the method on the `Notification` singleton class instead? One less module in the ancestor chains for the models, a bit more explicit API.